src/main.cpp

```diff
@@ -589,6 +603,14 @@ int main ( int argc, char** argv )
 //CTestbench Testbench ( "127.0.0.1", DEFAULT_PORT_NUMBER );
     // clang-format on
 
+    std::shared_ptr<CRpcServer> pRpcServer;
```

Comment:
I've not seen `std::shared_ptr<>` elsewhere in the code.  What does this provide that other methods of declaration do not?

Reply:
At first I was going to use `unique_ptr`, but the pointer is shared with `ClientRpc` and `ServerRpc` afterwards, so I use `shared_ptr` to make the intent more clear.

Reply:
I just read this: https://doc.qt.io/qt-5/objecttrees.html

I will see if I can use normal pointer and arrange objects this way:

- QApplication
  - CRpcServer
     - CClientRpc
     - CServerRpc

Reply:
Refactored the code to use Qt’s object trees.

Reply:
Just got segfault’d :joy:.

Apparently when declaring a smart pointer like `unique_ptr` or `shared_ptr`, it gets automatically initialized with null pointer.

But if I declare `CRpcServer* pRpcServer` without explicitly specifying `= nullptr`, it gets initialized with some random value.